go/enclave/main/main.go

```diff
@@ -2,28 +2,23 @@ package main
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/ten-protocol/go-ten/go/common/container"
-	tenflag "github.com/ten-protocol/go-ten/go/common/flag"
 	"github.com/ten-protocol/go-ten/go/config"
+	enclaveconfig "github.com/ten-protocol/go-ten/go/enclave/config"
 	enclavecontainer "github.com/ten-protocol/go-ten/go/enclave/container"
 )
 
 // Runs an Obscuro enclave as a standalone process.
 func main() {
-	// fetch and parse flags
-	flags := config.EnclaveFlags                       // fetch the flags that enclave requires
-	err := tenflag.CreateCLIFlags(config.EnclaveFlags) // using tenflag convert those flags into the golang flags package ( go flags is a singlen )
+	tenCfg, err := config.LoadTenConfig()
 	if err != nil {
-		panic(fmt.Errorf("could not create CLI flags. Cause: %w", err))
+		fmt.Println("Error loading ten config:", err)
+		os.Exit(1)
 	}
 
-	tenflag.Parse() // parse the golang flags package defined flags from CLI
-
-	enclaveConfig, err := config.NewConfigFromFlags(flags)
-	if err != nil {
-		panic(fmt.Errorf("unable to create config from flags - %w", err))
-	}
+	enclaveConfig := enclaveconfig.EnclaveConfigFromTenConfig(tenCfg)
```

Comment:
I don't follow here how the config is read from enclave.json

Reply:
Enclave config is read the same as the host and other processes: base yaml file(s) + environment variables

In enclave.json the "fromHost" flag allows you to whitelist env variable keys so their values can be passed in from the host machine.

E.g. if this was in the enclave.json then NETWORK_CHAINID=1337 would be set and ENCLAVE_DB_EDGELESSDBHOST value would be passed through if the runner set it as an environment variable.

```
"env": [
    { "name": "ENCLAVE_DB_EDGELESSDBHOST",  "fromHost": true },

    { "name": "NETWORK_CHAINID", value: 1337 },
]
```

So for now I'm just letting it pass through all the env variables. For tagged public releases we can hardcode values.

Reply:
what confuses me is that I don't see anywhere in the code how "fromHost" is resolved.
I don't follow how the right values end up in the enclave.json file that is passed to ego